command-handler/src/cmd-handler/command-handler.js

```diff
@@ -10,49 +15,85 @@ export default class CommandHandler {
     _commands = new Map();
 
     constructor(commandsDir, handler) {
+        //throw error if commands directory is not provided
         if (!commandsDir) throw new Error('Commands directory is required');
 
+        //initialize the command handler
+        //this function is called because the constructor cannot be async
         this.init(commandsDir, handler);
     }
 
+    //getter for commands
     get commands() {
         return this._commands;
     }
 
     async init(commandsDir, handler) {
         const registeredCommands = [];
+        //get all the files in the commands directory
```

Comment:
Some of these comments appear to have more characters than the code itself @NichArchA82